tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c

```diff
@@ -0,0 +1,476 @@
+{
+	if (settings_set_err) {
+		return settings_set_err;
+	}
+
+	return 0;
+}
+
+SETTINGS_STATIC_HANDLER_DEFINE(fp_reset_test, SETTINGS_RESET_TEST_SUBTREE_NAME, NULL,
+			       fp_reset_test_settings_set, fp_reset_test_settings_commit, NULL);
+
+static void mark_test_as_finished(enum reset_test_finished bit_pos)
+{
+	WRITE_BIT(test_finished_mask, bit_pos, 1);
+}
+
+static int test_fp_storage_reset_perform(void)
+{
+	if (storage_reset_pass) {
+		mark_test_as_finished(RESET_TEST_FINISHED_INTERRUPTED_RESET_BEGIN_BIT_POS);
+		storage_reset_pass = false;
+		ztest_test_pass();
+	}
+
+	return 0;
+}
+
+static void test_fp_storage_reset_prepare(void) {}
+
+/* The name starts with '_0_' so that this module's struct would be first in section (linker sorts
+ * entries by name) and its callbacks would be call before other's modules callbacks.
+ */
+FP_STORAGE_MANAGER_MODULE_REGISTER(_0_test_reset, test_fp_storage_reset_perform,
+				   test_fp_storage_reset_prepare);
+
+static void test_section_placement(void)
+{
+	struct fp_storage_manager_module *module;
+
+	STRUCT_SECTION_GET(fp_storage_manager_module, 0, &module);
+
+	zassert_equal_ptr(test_fp_storage_reset_perform, module->module_reset_perform,
+			  "Invalid order of elements in section");
+	zassert_equal_ptr(test_fp_storage_reset_prepare, module->module_reset_prepare,
+			  "Invalid order of elements in section");
+
+	mark_test_as_finished(RESET_TEST_FINISHED_SECTION_PLACEMENT_BIT_POS);
+}
+
+static int settings_validate_empty_cb(const char *key, size_t len, settings_read_cb read_cb,
+				      void *cb_arg, void *param)
+{
+	int *err = param;
+
+	*err = -EFAULT;
+	return *err;
+}
+
+static int settings_validate_empty(void)
+{
+	int err;
+	int err_cb = 0;
+
+	err = settings_load_subtree_direct(NULL, settings_validate_empty_cb, &err_cb);
+	if (err) {
+		return err;
+	}
+
+	if (err_cb) {
+		return err_cb;
+	}
+
+	return 0;
+}
+
+static int fp_data_nvs_validate_empty_cb(const char *key, size_t len, settings_read_cb read_cb,
+					      void *cb_arg, void *param)
+{
+	int *err = param;
+
+	if (!strncmp(key, SETTINGS_RESET_TEST_STAGE_FULL_NAME,
+		     sizeof(SETTINGS_RESET_TEST_STAGE_FULL_NAME))) {
+		return 0;
+	}
+
+	if (!strncmp(key, SETTINGS_RESET_IN_PROGRESS_FULL_NAME,
+		     sizeof(SETTINGS_RESET_IN_PROGRESS_FULL_NAME))) {
+		int rc;
+		bool reset_in_progress;
+
+		if (len > sizeof(reset_in_progress)) {
+			*err = -EINVAL;
+			return *err;
+		}
+
+		rc = read_cb(cb_arg, &reset_in_progress, len);
+		if (rc < 0) {
+			*err = rc;
+			return *err;
+		}
+
+		if (rc != len) {
+			*err = -EINVAL;
+			return *err;
+		}
+
+		if (reset_in_progress) {
+			*err = -EFAULT;
+			return *err;
+		}
+
+		return 0;
+	}
+
+	*err = -EFAULT;
+	return *err;
+}
+
+static int fp_data_nvs_validate_empty(void)
+{
+	int err;
+	int err_cb = 0;
+
+	err = settings_load_subtree_direct(NULL, fp_data_nvs_validate_empty_cb, &err_cb);
+	if (err) {
+		return err;
+	}
+
+	if (err_cb) {
+		return err_cb;
+	}
+
+	return 0;
+}
```

Comment:
Since you already erase storage area using `flash_area_erase`, you could use `flash_area_read` in `setup_fn` to ensure that each byte on FLASH area is equal to `flash_area_erased_val`. That would be more consistent approach I think. The function could be called e.g, `settings_area_validate_empty`
Then you could have a single function to validate that settings are empty (the current `fp_data_nvs_validate_empty`)

Reply:
Discussed offline. We need to change the approach to make sure that settings partition is properly initialized before each unit test.